src/utils/codeBuilder.ts

```diff
@@ -0,0 +1,309 @@
+    ) {
+        // ensure dirPath exists
+        if (!fs.existsSync(dirPath)) {
+            try {
+                fs.mkdirSync(dirPath, { recursive: true });
+            } catch (err) {
+                console.log(`Unable to create directory: ${dirPath}`, err);
+                return;
+            }
+        }
+        // write the file
+        const filePath = path.join(dirPath, filename);
+        try {
+            fs.writeFileSync(filePath, content, 'utf8');
+        } catch (err) {
+            console.error(`Error writing to file ${filePath}`, err);
+        }
+    }
+
+    /**
+     * Ensure all the TEMPLATE_* variables have a value.
+     */
+    private generateTemplateVariables(fieldNames: string[]): TemplateVariables {
+        const templateVariables: TemplateVariables = {};
+        templateVariables[CodeBuilder.TEMPLATE_OBJECT_API_NAME] =
+            this.objectApiName;
+
+        // Labels
+        templateVariables[
+            CodeBuilder.TEMPLATE_CREATE_LWC_LABEL
+        ] = `LWC for creating a/an ${this.objectApiName} instance.`;
+        templateVariables[
+            CodeBuilder.TEMPLATE_EDIT_LWC_LABEL
+        ] = `LWC for editing a/an ${this.objectApiName} instance.`;
+        templateVariables[
+            CodeBuilder.TEMPLATE_VIEW_LWC_LABEL
+        ] = `LWC for viewing a/an ${this.objectApiName} instance.`;
+
+        // We need to populate the following template variables:
+        // TEMPLATE_FIELDS - a comma separated list of field names from the import statements, used in viewRecord template.
+        //    ie: return [NAME_FIELD, PHONE_FIELD, WEBSITE_FIELD, INDUSTRY_FIELD, TYPE_FIELD];
+        // TEMPLATE_IMPORTS - a list of import statements that pulls in the @salesforce/schema fields:
+        //    ie: import NAME_FIELD from "@salesforce/schema/Account.Name";
+        // TEMPLATE_LIGHTNING_INPUT_CREATE_FIELDS_HTML - fields specified as lightning-input-field values in the create html:
+        //    ie: <lightning-input-field field-name={nameField} value={name}></lightning-input-field>
+        // TEMPLATE_LIGHTNING_INPUT_EDIT_FIELDS_HTML - fields specified as lightning-input-field values in the edit html
+        //    ie: <lightning-input-field field-name={nameField}></lightning-input-field>
+        // TEMPLATE_VARIABLES - aliases the imported fields to variables
+        //    ie: nameField = NAME_FIELD;
+        // TEMPLATE_VARIABLE_ASSIGNMENTS - stores the value of the create fields:
+        //    ie: name = "";
+
+        var fields = '';
+        var imports = '';
+        var createFieldsHtml = '';
+        var editFieldsHtml = '';
+        var importAliases = '';
+        var variableAssignments = '';
+
+        fieldNames.forEach((field) => {
+            var fieldNameImport = `${field.toUpperCase()}_FIELD`;
+            fields += `${fieldNameImport}, `;
+            imports += `import ${fieldNameImport} from "@salesforce/schema/${this.objectApiName}.${field}";\n`;
```

Comment:
Do you think it would be worthwhile to represent these template strings as constants at the top of the file? So that you'd have something like:

```typescript
const IMPORT_STATEMENT_TEMPLATE = 'import <FieldNameImport> from "@salesforce/schema/<ObjectApiName>.<FieldName>";'
const LIGHTNING_INPUT_FIELD_TEMPLATE = '<lightning-input-field field-name={<FieldNameVariable>} value={<FieldName>}></lightning-input-field>';
```
And then do substitutions here? I can't decide if that would make things more readable or less readable.

Reply:
Yeah, I'm not sure either. I am not a fan of all this string replacement, but not sure what else to do. Since it is hard to read already, moving the strings someplace else I almost feel like it could be more difficult to debug and/or maintain. But, I'm up for whatever!

Reply:
> moving the strings someplace else I almost feel like it could be more difficult to debug and/or maintain

That was my concern too. I'm okay if we leave it, unless I think of some far more compelling way to make it more readable—which I haven't come up with yet. 😛